src/benchmark/java/com/commercetools/sync/benchmark/ProductTypeSyncBenchmark.java

```diff
@@ -0,0 +1,286 @@
+                .withFailMessage(format("Diff of benchmark '%e' is longer than expected threshold of '%d'.",
+                        diff, THRESHOLD))
+                .isLessThanOrEqualTo(THRESHOLD);
+
+        // Assert actual state of CTP project (number of updated product types)
+        final CompletableFuture<Integer> totalNumberOfProductTypesWithOldName =
+                CTP_TARGET_CLIENT.execute(ProductTypeQuery.of()
+                                                          .withPredicates(
+                                                              p -> p.attributes().name().is("attr_name_1_old")))
+                                 .thenApply(PagedQueryResult::getTotal)
+                                 .thenApply(Long::intValue)
+                                 .toCompletableFuture();
+
+        executeBlocking(totalNumberOfProductTypesWithOldName);
+        assertThat(totalNumberOfProductTypesWithOldName).isCompletedWithValue(0);
+
+        // Assert actual state of CTP project (total number of existing product types)
+        final CompletableFuture<Integer> totalNumberOfProductTypes =
+                CTP_TARGET_CLIENT.execute(ProductTypeQuery.of())
+                                 .thenApply(PagedQueryResult::getTotal)
+                                 .thenApply(Long::intValue)
+                                 .toCompletableFuture();
+        executeBlocking(totalNumberOfProductTypes);
+        assertThat(totalNumberOfProductTypes).isCompletedWithValue(NUMBER_OF_RESOURCE_UNDER_TEST);
+
+        // Assert statistics
+        AssertionsForStatistics
+                .assertThat(syncStatistics)
+                .hasValues(NUMBER_OF_RESOURCE_UNDER_TEST, halfNumberOfDrafts, halfNumberOfDrafts, 0);
+
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+        assertThat(warningCallBackMessages).isEmpty();
+
+        saveNewResult(SyncSolutionInfo.LIB_VERSION, PRODUCT_TYPE_SYNC, UPDATES_ONLY, totalTime);
+    }
+
+
+    @Nonnull
+    private static List<ProductTypeDraft> buildProductTypeDrafts(final int numberOfTypes) {
+        return IntStream
+                .range(0, numberOfTypes)
+                .mapToObj(i -> ProductTypeDraftBuilder.of(
+                        format("key__%d", i),
+                        format("name__%d", i),
+                        format("description__%d", i),
+                        singletonList(ATTRIBUTE_DEFINITION_DRAFT_1)
+                ).build())
+                .collect(Collectors.toList());
+    }
+
+    @Nonnull
+    private static ProductTypeDraftBuilder applyAttributeDefinitionNameChange(
+            @Nonnull final ProductTypeDraftBuilder builder) {
+
+        final List<AttributeDefinitionDraft> list =
+                builder.getAttributes()
+                       .stream()
+                       .map(attributeDefinitionDraft -> AttributeDefinitionDraftBuilder
+                               .of(
```

Comment:
@ahmetoz wouldn't be easier to do it like this: 
```suggestion
AttributeDefinitionDraftBuilder.of(attributeDefinitionDraft)
                               .name(attributeDefinitionDraft.getName() + "_old")
                               .build()
```
as opposed to copying every single value?

Reply:
oh, nice! 
https://github.com/commercetools/commercetools-sync-java/pull/317/commits/e7fca684b45052ddcdb1bdb57d5d9f30f1898018